CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+
+Feature requests are always welcome but many times the answer is:
+
+> That's a great idea, you should write a plug-in for that! It doesn't belong in the core library and it's probably something you could do with only a small amount of code.
+
+If you'd like to contribute a feature you should always first open an issue to discuss if the feature belongs in core vs a plug-in.  Often this is a great way to get tips for implemention as well or links to prior disussions with context, etc.
+
+Some bullet points:
+
+- Our [lack of line numbers is a feature](https://highlightjs.readthedocs.io/en/latest/line-numbers.html). The simple case isn't hard to add though with a little custom JavaScript.
+- We [embrace plug-ins and extensions](https://github.com/highlightjs/highlight.js/issues/2225) rather than adding lots of tiny features and configuration options.
+- We are much more than a keyword highlighter. We do make attempts to understand context...
+- ...Yet we are not a full parser for any language, we're still just a fancy nested pattern matcher.
+- Our [auto-detect is sadly not powered by magic unicorn dust](https://github.com/highlightjs/highlight.js/issues/1213). It's best effort, not best in class.
+
+
+## Reporting an Issue
+(Misc.)
+- If you found a language highlighting issue, use [this JSFiddle](https://jsfiddle.net/ajoshguy/2bmdswn6/) to create a test case.
+
+
+## Fixing an Issue (PRs)
+If you're new to contributing to open-source, have a look at [this GitHub Guide](https://guides.github.com/activities/forking). It explains the general process of GitHub collaboration.
+
+If you feel comfortable with our [prerequisites](#prerequisites), you can grab any issue [marked `beginner friendly`](https://github.com/highlightjs/highlight.js/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22).
+
+
+### Rules
+Don't worry; these rules are pretty standard.
+
+Please open an issue before your PR, so we can explore and discuss the topic at hand. Your time is important, and we need to be sure it's well-spent.
+
+**Before** you start coding, keep these tips in mind:
+- You should usually add markup tests (ie. whenever you've made a significant grammar change or fixed a bug)
+- Change only what needs to be changed; don't re-lint/rewrite whole files when fixing bugs
+- Linting needs a dedicated commit
+
+**When you're ready** to open a PR, we have some housekeeping tasks for you.
+- Find and link to the relevant issue
+- Add a changelog entry to [CHANGES.md](https://github.com/highlightjs/highlight.js/blob/master/CHANGES.md)
+- If necessary, add your name to [AUTHORS.txt](https://github.com/highlightjs/highlight.js/blob/master/AUTHORS.txt) under Contributors
```

Comment:
Is this redundant now that it's in our PR template or is this just good reinforcing?

Reply:
I meant it as reinforcement. We could remove it, if you want.

Reply:
Feels a bit redundant... (the list)... It could just say "when you open a PR we'll have a small checklist waiting for you... it's easy".  No need to repeat.